File: 215_kth_largest.java
// https://leetcode.com/problems/kth-largest-element-in-an-array/description/
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> q = new PriorityQueue<>((a, b) -> Integer.compare(b, a));

        for (int n: nums) {
            q.add(n);
        }

        int n=0;
        for (int i = 0; i < k; i++){
            n = q.poll();
        }

        return n;
    }
}

File: 215_kth_largest.py
# https://leetcode.com/problems/kth-largest-element-in-an-array/description/
import heapq 

class Solution:
    def solve(self, nums: list[int], k: int) -> int:
        heapq.heapify(nums)
        return heapq.nlargest(k, nums)[-1]


def main():
    s = Solution()
    print(s.solve([3,2,1,5,6,4], 2))

if __name__ == '__main__':
    main()